src/DIG/Dialog.jsx

```diff
@@ -202,7 +202,7 @@ return (
         )}
       </ActionWrapper>
       {(type === "dialog" || enableCloseButton) && (
-        <CloseButtonWrapper>
+        <CloseButtonWrapper onClick={onCancel}>
```

Comment:
```suggestion
        <CloseButtonWrapper>
```

No need to pass another prop here. Just use [onOpenChange](https://www.radix-ui.com/primitives/docs/components/alert-dialog#root) in `ReasonDialog` component